src/Config.php

```diff
@@ -71,50 +74,62 @@
  * For classes known from the definitions, a type preference might be the
  * better approach
  *
- * @see Laminas\Di\Resolver\ValueInjection A container to force injection of a value
- * @see Laminas\Di\Resolver\TypeInjection  A container to force looking up a specific type instance for injection
+ * @see \Laminas\Di\Resolver\ValueInjection A container to force injection of a value
+ * @see \Laminas\Di\Resolver\TypeInjection  A container to force looking up a specific type instance for injection
+ *
+ * @psalm-type TypeConfigArray = array{
+ *  typeOf?: class-string|null,
```

Comment:
Do we actually allow `null` or is that something like a real-world example?

I would say, if `typeOf` is provided (thanks to `?`, it is not a required array key), only `class-string` should be provided. 🤷🏼‍♂️

Reply:
That depends. When we want to provide a way to remove existing settings from a ConfigProvider or project level config file, then allowing `null` is the only option, since the Laminas `config` array is merged from config providers or MVC modules.

I am not sure if this method this is actively used, but it is a valid possibility at the moment.